### PR TITLE
Updating Documentation

### PR DESCRIPTION
**Pull request by beautifulentropy**
Changed all mentions of master to main, fixed some wording and ordering,
word-wrapped, and added Prerequisites section that contains some useful
documentation links.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,112 +1,207 @@
 # Boulder Release Process Docs & Demo
 
 A repo for documenting and demonstrating the
-[Boulder](https://github.com/letsencrypt/boulder) release process used to move
-code from Github to [staging](https://letsencrypt.org/docs/staging-environment/)
-and [production](https://acme-v01.api.letsencrypt.org/).
+[Boulder](https://github.com/letsencrypt/boulder) release process used
```
Comment (aarongable) [inline README.md:4]: Looks like the wrapping wrapped to significantly less than 80 characters (was it accidentally set to 72?).

More generally, re-wrapping files like this is a great cleanup to do but it's best for it to be in a separate commit from content changes, so that it's easier to tell when substantive things changed vs just style things.

File: README.md [hunk 2]
```diff
@@ -1,112 +1,207 @@
...
 
-* After deploying `release-2018-01-30` a critical bug is discovered! Since `master` is clean, that is, has no commits since `release-2018-01-30` we don't want to ship with a hotfix, @cpu quickly prepares a [hotfix PR](https://github.com/letsencrypt/boulder-release-process/pull/2) to propose a hotfix change be merged into `master`.
+* The signing GPG key must be added to your GitHub account:
+  * [Adding a new GPG key to your GitHub
+    account](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/adding-a-new-gpg-key-to-your-github-account)
 
-* @jsha reviews and [approves the hotfix PR](https://github.com/letsencrypt/boulder-release-process/pull/2#pullrequestreview-92737652). The corresponding [travis build](https://travis-ci.com/letsencrypt/boulder-release-process/builds/64675859?utm_source=github_status&utm_medium=notification) passes, and so [@cpu merges the hotfix PR](https://github.com/letsencrypt/boulder-release-process/commit/4ed5f8181852237dd6cde45fbc63bd199f2d3ec9) into master.
+* `git` must be configured to call the correct GPG binary:
+  * On most Linux platforms: `git config --global gpg.program gpg`
```
Comment (aarongable) [inline README.md:107]: I believe this is the default, and isn't necessary most of the time.

Comment (beautifulentropy) [reply README.md:107]: I had somehow set mine to gpg2 and that was not right, I don't want someone else to experience the same pain of figuring that out if it can be helped.

File: README.md (hunk 1 above)
Comment (beautifulentropy) [reply README.md:4]: Ah, yeah, my bad on both counts, I'll fix it.

File: README.md (hunk 2 above)
Comment (beautifulentropy) [reply README.md]: I have reworded this. If it still doesn't make sense to keep it, I can remove it.

**Review by aarongable — APPROVED** [review]

**Review by jsha — APPROVED** [review]

**Review by aarongable — COMMENTED** [review]
Re-approving now that I actually can?
